RocketRL/python/func/data_processing.py

```diff
@@ -13,48 +13,83 @@
 import statsmodels.api as sm
 from numpy import ones,vstack
 from numpy.linalg import lstsq
-
-import yaml
+import pickle
 import os
 
-#set config path
-config_dir = 'config/'
-CWD_PATH = os.getcwd()
-config_path = os.path.join(CWD_PATH,config_dir,"data.yml")
-
-with open(config_path, 'r') as ymlfile:
-    cfg = yaml.load(ymlfile)
+# Import function to create training and test set splits
+from sklearn.cross_validation import train_test_split
+# Import function to automatically create polynomial features! 
+from sklearn.preprocessing import PolynomialFeatures
+# Import Linear Regression and a regularized regression function
+from sklearn.linear_model import LinearRegression
+from sklearn.linear_model import LassoCV
+# Finally, import function to make a machine learning pipeline
+from sklearn.pipeline import make_pipeline
 
-#TO DO: is below kosher?
 
-all_var = cfg['in_var'].copy()
-for item in cfg['out_var']:
-    all_var.append(item)
-cfg['all_var']= all_var
+from PIL import Image
+from PIL import ImageFont
+from PIL import ImageDraw  
+from PIL import ImageOps
 
+import cv2 as cv
+from cv2 import imwrite
 
+import pyglet
+#TO DO:
 #class RocketData():
 
-#clear workspace
-plt.close('all')
+def labelmaker(cfg,labels):
+    #to do: get these to show up in rendering env
+    save_dir = os.path.join(cfg['CWD_PATH'], cfg['labels_path'])
+    
+    for fname, label in cfg[labels].items():
+        if labels == 'labels1':
+            lbllen = len(label)
+        else: 
+            lbllen = 1
+            label = str(label)
+        width = lbllen*cfg['label_mult']
+        height = cfg['label_height']
+        
+#        fig = plt.figure(figsize = (width/50,height/100))
+#        ax = fig.add_subplot(111)
+#        ax.annotate(label,(0,0),size= 16)
+#        fig.savefig(save_dir+fname+cfg['ftype'])
+        
+#        img = np.zeros((height,width,3), np.uint8)
+#        font = font = cv.FONT_HERSHEY_SIMPLEX
+#        cv.putText(img, label, (0, 20),font, 0.8, (0, 255, 0), 2, cv.LINE_AA)
+#        imwrite( save_dir+fname+cfg['ftype'], img )
+#        
+##        
+##        
+        img = Image.new('RGBA', (width, height),"white")
+        draw = ImageDraw.Draw(img)
+        font = ImageFont.truetype("Arial.ttf", 18) 
+        # draw.text((x, y),"Sample Text",(r,g,b))
+        draw.text((0, 0),label,(100,100,100),font=font)#(255,255,255))#
+#        img = ImageOps.invert(img)
+        f_name = save_dir+fname+cfg['ftype']
+        img.save(f_name)
+        kitten = pyglet.image.load(f_name)
+        kitten.save(f_name)
 
-def plotter(plotd):
-    if '2d' in plotd:
-#        self.plotting2D = True
-        plotting2D = True
-    if '3d' in plotd:
-#        self.plotting3D = True
-        plotting3D = True
+def cfg_mod(cfg):
+     #TO DO: is below kosher?
+    all_var = cfg['in_var'].copy()
+    for item in cfg['out_var']:
+        all_var.append(item)
+    cfg['all_var']= all_var
+    return cfg
```

Comment:
Is there a reason you decided to merge both `in_var` and `out_var` inside the `all_var`? Couldn't you just access them by using `cfg['in_var']['example_var']`?

Reply:
that gives me an error message, list indices must be integers or slices, not str
[cfg['in_var'], cfg['out_var']] is the wrong shape